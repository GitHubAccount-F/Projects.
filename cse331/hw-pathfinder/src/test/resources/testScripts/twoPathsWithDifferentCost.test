#Tests Dijkstra's algorithm on a graph with two different paths to
#a target node, with each path have a different cost.

#Create graph
CreateGraph graph1

#Add nodes
AddNode graph1 one
AddNode graph1 two
AddNode graph1 three
AddNode graph1 four

#Adds edges
AddEdge graph1 one two 5.25
AddEdge graph1 two three 5.25
AddEdge graph1 three four 5.25
AddEdge graph1 one four 12.67
AddEdge graph1 one three 20

#Finds the least costly path
FindPath graph1 one four

FindPath graph1 one three