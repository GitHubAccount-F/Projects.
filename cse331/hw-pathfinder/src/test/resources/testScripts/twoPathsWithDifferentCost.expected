#Tests Dijkstra's algorithm on a graph with two different paths to
#a target node, with each path have a different cost.

#Create graph
created graph graph1

#Add nodes
added node one to graph1
added node two to graph1
added node three to graph1
added node four to graph1

#Adds edges
added edge 5.250 from one to two in graph1
added edge 5.250 from two to three in graph1
added edge 5.250 from three to four in graph1
added edge 12.670 from one to four in graph1
added edge 20.000 from one to three in graph1

#Finds the least costly path
path from one to four:
one to four with weight 12.670
total cost: 12.670

path from one to three:
one to two with weight 5.250
two to three with weight 5.250
total cost: 10.500